static/js/tools/choropleth/choropleth.tsx

```diff
@@ -450,6 +542,45 @@ class ChoroplethMap extends Component<PropsType, StateType> {
     }
   };
 
+  /**
+   * Returns the complete select/dropdown component
+   * with the available dates as options.
+   */
+  public datePicker = (): JSX.Element => {
+    // Dates are not repeated, so store them as a set.
+    const allDates: Set<string> = new Set();
+    Object.entries(this.state["data"])?.forEach(([, dateToValue]) => {
+      const dates = Object.keys(dateToValue);
+      dates.forEach((date) => {
+        allDates.add(date);
+      });
+    });
+
+    // Sort the dates in descending order.
+    const sortedDates = Array.from(allDates).sort().reverse();
+
+    // Create the option components.
+    const dateComponents = sortedDates.map((date) => {
+      return (
+        <option key={date} value={date}>
+          {date}
+        </option>
+      );
+    });
+
+    // Return the select component.
+    return (
+      <select onChange={this.changeDate} defaultValue={this.state["date"]}>
```

Comment:
it's good practice to keep the handler function names of the form "onDateChanged" or "onSelectedDateChanged"

Reply:
fixed.